clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -338,7 +341,15 @@ class ScalarExprEmitter : public StmtVisitor<ScalarExprEmitter, mlir::Value> {
     } else if (type->isVectorType()) {
       llvm_unreachable("no vector inc/dec yet");
     } else if (type->isRealFloatingType()) {
-      llvm_unreachable("no float inc/dec yet");
+      auto isFloatOrDouble = type->isSpecificBuiltinType(BuiltinType::Float) ||
+                             type->isSpecificBuiltinType(BuiltinType::Double);
+      assert(isFloatOrDouble && "NYI");
```

Comment:
nit: more detailed message for the assert? e.g, "Non-float/double NYI".

Reply:
Applied!